java/core/src/main/java/co/worklytics/psoxy/impl/RESTApiSanitizerImpl.java

```diff
@@ -259,12 +271,72 @@ public MapFunction getTransformImpl(Transform transform) {
             f = getFilterTokenByRegex((Transform.FilterTokenByRegex) transform);
         } else if (transform instanceof Transform.Tokenize) {
             f = getTokenize((Transform.Tokenize) transform);
+        } else if (transform instanceof HashIp) {
+            f = getHashIp((HashIp) transform);
+        } else if (transform instanceof EncryptIp) {
+            f = getEncryptIp((EncryptIp) transform);
         } else {
             throw new IllegalArgumentException("Unknown transform type: " + transform.getClass().getName());
         }
         return f;
     }
 
+    @VisibleForTesting
+    MapFunction getEncryptIp(EncryptIp transform) {
+        return (s, jsonConfiguration) -> {
+            if (!(s instanceof String)) {
+                if (s != null) {
+                    log.warning("value matched by " + transform + " not of type String");
+                }
+                return null;
+            } else if (StringUtils.isBlank((String) s)) {
+                return s;
+            } else {
+                try {
+                    // Parse the IP address string and convert back to string to get the canonical form.
+                    InetAddress address = InetAddress.getByName((String) s);
```

Comment:
this is going to do an extra call, but OK as it has a cache built-in.

Reply:
But the cache is local, as part of the static instance of `InetAddress`, of course this is better than nothing, but if worth to use `ConfigService` to store the value and use it as a shared cache?

Reply:
I think I removed this, as I decided preserving format not worthwhile.